### DISCUSSION TITLE
How I can  pass environment variables from a Next.js app to packages in a turbo repository?

### DISCUSSION THREAD

**Discussion by kuji2336** [vercel/turborepo#4970] [Help] [answered]
I have a package named "test-package" in the "packages" folder. It should use the .env variables from where it is imported. In my case, it is imported in the "project1" app, which is a Next.js application. When I import the "test-package" into the "project1" application and check it, it can read the .env variables from the "project1" application without any additional changes. Is this correct way to access .env variables in package?

**Comment by anthonyshew** [comment, marked as answer]
You'll want to think of your application as "the thing that is running" and a package as "a part of the thing that is running."

For what you're describing, the environment variables will be available to the package because it is in the runtime of the application. If your package has self-runnable scripts, those scripts will be run in their own runtime and would need the variables assigned to that runtime.

An example for thinking about this would be a `database` package that:
- exports a client to use in applications
- can run migrations (outside of the context of the application)

When you use the client in an application, the environment variables are a part of the applicaiton's runtime environment so the client will use them. But, when you run a migration, the `database` package needs environment variables given to that migration's runtime (since it's running in the application).
